### ISSUE TITLE
Numpad under Wayland always acts as if disabled (numlock off)

### ISSUE THREAD

**Issue by SkewedZeppelin** [libsdl-org/SDL#4666]
versions:
- SDL2-2.0.16-1.fc34 with SDL_VIDEODRIVER=wayland
- mutter-40.3-1.fc34
- gnome-shell-40.3-1.fc34
- Fedora 34 GNOME with Wayland

Numpad acts like numlock is off, so eg. enter key is ENTER instead of KP_ENTER.

I cannot reproduce under X driver (default), or under 2.0.14.

My likely (wrong) guess is related to 74162b7401844d58ee82541b39c51ec2aa264867

**Comment by flibitijibibo** [comment]
Probably a duplicate of #4642

**Comment by flibitijibibo** [comment]
I've been running through this one for a couple hours since it's the easiest thing to fix in isolation (basically just KP_ENTER), and I'm at a bit of a loss because it seems like xkbcommon diverges from X pretty hard here.

Basically the mess works like this for both X and Wayland:

1. Verify the SDL_Scancode is sane for the given keycode
2. Translate the keycode to a keysym
3. Translate the keysym to an SDL_Keycode, map to the SDL_Scancode

Everything up to step 3 is identical, and in fact step 3 is _supposed_ to be identical as well. It comes down to this: `KeySymToUcs4` is incompatible with `xkb_keysym_to_utf32`, which is mildly annoying because the [xkbcommon docs](https://xkbcommon.org/doc/current/group__keysyms.html#ga198462d79271d614c43040595bbe2364) plainly state that it should be compatible, but sadly the X function does _just_ a bad enough job to where the keymap logic works.

In the case of KP_ENTER, both X and xkbcommon get a unique KP_ENTER sym, but when passed to X the Ucs4 return value is 0, which then falls back to the standard scancode->keycode conversion and allows the binding to work. Wayland, however, translates to '\13`, which isn't _wrong_ but makes it completely impossible for us to differentiate Enter from KP Enter using this function!

Basically the only way we'll be able to escape from this and #4642 is to make a giant dictionary of the values in xkbcommon-keysyms.h and SDL_keycode.h, bypassing the xkbcommon function entirely.
